iidm/iidm-api/src/main/java/com/powsybl/iidm/network/ShuntCompensatorNonLinearModel.java

```diff
@@ -0,0 +1,54 @@
+/**
+ * Copyright (c) 2020, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+package com.powsybl.iidm.network;
+
+import java.util.List;
+import java.util.Optional;
+
+/**
+ * @author Miora Ralambotiana <miora.ralambotiana at rte-france.com>
+ */
+public interface ShuntCompensatorNonLinearModel extends ShuntCompensatorModel {
+
+    interface Section {
+
+        /**
+         * Get the susceptance in S of the section.
+         */
+        double getB();
+
+        /**
+         * Get the conductance in S of the section.
+         */
+        double getG();
+    }
+
+    /**
+     * Get an optional of the section associated with a given section number if it exists.
+     * If such a section does not exist, return an empty optional.
+     *
+     */
+    Optional<Section> getSection(int sectionIndex);
+
+    /**
+     * Get all the sections associated with their section number.
+     */
+    List<Section> getSections();
+
+    /**
+     * For a given section number, add a section with a given susceptance and conductance in S to the model.
+     * If a section already exists for this section number, respectively replace its susceptance and conductance with the given susceptance and conductance.
+     * Throw an exception if the section index equals to 0 (corresponds to disconnected state).
+     */
+    ShuntCompensatorNonLinearModel addOrReplaceSection(int sectionIndex, double b, double g);
+
+    /**
+     * Remove the section associated with a given section number if it exists <b>and</b> the current section count is different of the given section number.
+     * Else, throw an exception.
+     */
+    ShuntCompensatorNonLinearModel removeLastSection();
```

Comment:
Why do you need this function ?

Reply:
Otherwise, it would be impossible to remove shunt section in non linear shunt (for development purposes for example)

Reply:
I am not sure to get what you mean :-) The function is not to remove a given section, just the last one.

Reply:
When you have a non linear shunt, what you can want to do is:
- adding a new section
- replacing the value of an existing section
- removing an existing section
However, when you remove an existing section, it cannot be one in the middle (you cannot have a "hole" in your section list), so it is necessarily the last section. It occurs to me that a method can however be added to remove the last 2, 3, etc sections of a non linear shunt and not only the last one... (for now, to do that we have to call removeLastSection 2, 3, etc times)

Reply:
Side note, it is the same for adding a new section, you cannot add a section with a index strictly superior to the last index + 1 to avoid "holes"

Reply:
Thanks so Indeed this remove function is using by the function addOrReplace ? Why have you decided to put it in the interface and not just in the implementation?

Reply:
No it is used when we really want to remove a section! addOrReplace can not remove a section, only replacing a value or adding a section.

Reply:
I don't understand why is a problem to remove a section that is not the last.
In any case if you remove a section it doesn't make sense anymore to update back the original model so we can allow everything.

Reply:
Okay, just a question: when a section is removed, should all the cumulated susceptances and admittances of following sections be calculated again or does it just mean it will not be possible anymore to activate a certain section count to access a given cumulated susceptance and admittance?

For example, if we have:
```xml
<shunt id="SHUNT">
  <shuntNonLinearModel>
    <section b="1.0" g="1.0"/>
    <section b="2.0" g="2.0"/>
    <section b="4.0" g="5.0"/>
  </shuntNonLinearModel>
</shunt>
```
If the section section is removed, should we have:
```xml
<shunt id="SHUNT">
  <shuntNonLinearModel>
    <section b="1.0" g="1.0"/>
    <section b="4.0" g="5.0"/>
  </shuntNonLinearModel>
</shunt>
```
or:
```xml
<shunt id="SHUNT">
  <shuntNonLinearModel>
    <section b="1.0" g="1.0"/>
    <section b="3.0" g="4.0"/>
  </shuntNonLinearModel>
</shunt>
```
?

Reply:
I guess my question should be: do you know the use case where a section is "removed" (for maintenance, etc) and what is the behavior of the shunt in this case?